SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp

```diff
@@ -105,61 +105,102 @@ bool WriteLoadbalancingSection(TSharedRef<TJsonWriter<>> Writer, const FName& Wo
 
 } // anonymous namespace
 
-void SetLevelEditorPlaySettingsWorkerTypes(const TMap<FName, FWorkerTypeLaunchSection>& InWorkers)
+void SetLevelEditorPlaySettingsWorkerType(const FWorkerTypeLaunchSection& InWorker)
 {
 	ULevelEditorPlaySettings* PlayInSettings = GetMutableDefault<ULevelEditorPlaySettings>();
 
-	PlayInSettings->WorkerTypesToLaunch.Empty(InWorkers.Num());
+	PlayInSettings->WorkerTypesToLaunch.Empty(1);
 
-	if (InWorkers.Num() == 0)
+	// TODO: Engine PR to remove PlayInSettings WorkerType map.
+	PlayInSettings->WorkerTypesToLaunch.Add(SpatialConstants::DefaultServerWorkerType, InWorker.NumEditorInstances);
+}
+
+uint32 GetWorkerCountFromWorldSettings(const UWorld& World)
+{
+	const ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(World.GetWorldSettings());
+
+	if (WorldSettings == nullptr)
 	{
-		UE_LOG(LogSpatialGDKDefaultLaunchConfigGenerator, Warning, TEXT("No workers specified in SetLevelEditorPlaySettingsWorkerType."));
+		UE_LOG(LogSpatialGDKDefaultLaunchConfigGenerator, Error, TEXT("Missing SpatialWorldSettings on map %s"), *World.GetMapName());
+		return 1;
 	}
 
-	for (const auto& Worker : InWorkers)
+	if (WorldSettings->bEnableMultiWorker == false)
 	{
-		if (Worker.Value.bAutoNumEditorInstances)
+		return 1;
+	}
+
+	FSpatialGDKEditorModule& EditorModule = FModuleManager::GetModuleChecked<FSpatialGDKEditorModule>("SpatialGDKEditor");
+	uint32 NumWorkers = 0;
+	if (WorldSettings->DefaultLayerLoadBalanceStrategy == nullptr)
+	{
+		UE_LOG(LogSpatialGDKDefaultLaunchConfigGenerator, Error, TEXT("Missing Load balancing strategy on map %s"), *World.GetMapName());
+		return 1;
+	}
+	else
+	{
+		UAbstractRuntimeLoadBalancingStrategy* LoadBalancingStrat = nullptr;
+		FIntPoint Dimension;
+		if (!EditorModule.GetLBStrategyExtensionManager().GetDefaultLaunchConfiguration(WorldSettings->DefaultLayerLoadBalanceStrategy->GetDefaultObject<UAbstractLBStrategy>(), LoadBalancingStrat, Dimension))
```

Comment:
We may want to make a different accessor to just return the needed number of workers, but this PR was already so large that I just reused the GetDefaultLaunchConfiguration even though it gets way more information than needed.